src/main/java/mikera/vectorz/impl/SparseIndexedVector.java

```diff
@@ -145,8 +147,59 @@ public static AVector createFromRow(AMatrix m, int row) {
 	public int nonSparseElementCount() {
 		return data.length;
 	}
-	
-	@Override
+
+    public AVector innerProduct(SparseRowMatrix m) {
+		int cc = m.columnCount();
+		int rc = m.rowCount();
+		checkLength(rc);
+        ASparseIndexedVector r = SparseIndexedVector.createLength(cc);
+		for (int i = 0; i < rc; i++) {
+            AVector row = m.getRow(i);
+            if (row instanceof ZeroVector) {
+                continue;
+            }
```

Comment:
We don't guarantee sparse return types in all cases. For small-constant-sized vectors, one of the primitive vectors or a simple dense Vector is definitely going to be better than a sparse type.

Haven't fully investigated where the cutoff is.